Support generation of Job/CronJob resources (#27425)
quarkusio/quarkus: docs/src/main/asciidoc/deploying-to-kubernetes.adoc

The image that will be used in the generated manifests will be `quarkus/demo-app:1.0`

=== Changing the generated deployment resource

> Beside generating a `Deployment` resource, you can also choose to get either a `StatefulSet`, or a `Job`, or a `CronJob` resource instead via `application.properties`:

Reviewer: ```suggestion
Besides generating a `Deployment` resource, you can also choose to generate either a `StatefulSet`, or a `Job`, or a `CronJob` resource instead via `application.properties`:
```
`beside` and `besides` both exists but given it's either one or another, I think it should be `besides`.

[source,properties]
----
quarkus.kubernetes.deployment-kind=StatefulSet
----

==== Generating Job resources

> If you want to generate a Job resource, you need to add the following property via the `application.properties`:

Reviewer: ```suggestion
If you want to generate a Job resource, you need to add the following property to the `application.properties`:
```

[source,properties]
----
quarkus.kubernetes.deployment-kind=Job
----

> IMPORTANT: If you are using the Picocli extension, by default the Job resource will be generated.

Reviewer: ```suggestion
IMPORTANT: If you are using the Picocli extension, by default a Job resource will be generated.
```

> You can provide the arguments that will be used by the Kubernetes Job via the property `quarkus.kubernetes.arguments`. For example, adding the property `quarkus.kubernetes.arguments=A,B`.

Reviewer: ```suggestion
You can provide the arguments that will be used by the Kubernetes Job via the property `quarkus.kubernetes.arguments`. For example, by adding the property `quarkus.kubernetes.arguments=A,B`.
```

> Finally, the Kubernetes job will be launched every time that is installed in Kubernetes. You can know more about how to run Kubernetes jobs in this https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job[link].

Reviewer: ```suggestion
Finally, the Kubernetes job will be launched every time it is installed in Kubernetes. You can know more about how to run Kubernetes jobs in this https://kubernetes.io/docs/concepts/workloads/controllers/job/#running-an-example-job[link].
```

> You can configure the rest of the Kubernetes Job configuration using the properties under `quarkus.kubernetes.job.xxx` (see https://quarkus.io/guides/deploying-to-kubernetes#quarkus-kubernetes-kubernetes-config_quarkus.kubernetes.job.parallelism-parallelism[link]).

Reviewer: ```suggestion
You can configure the rest of the Kubernetes Job configuration using the properties under `quarkus.kubernetes.job.xxx` (see the https://quarkus.io/guides/deploying-to-kubernetes#quarkus-kubernetes-kubernetes-config_quarkus.kubernetes.job.parallelism-parallelism[Deploying to Kubernetes guide] for more information).
```

==== Generating CronJob resources

> If you want to generate a CronJob resource, you need to add the following property via the `application.properties`:

Reviewer: ```suggestion
If you want to generate a CronJob resource, you need to add the following properties to the `application.properties`:
```

[source,properties]
----
quarkus.kubernetes.deployment-kind=CronJob
# Cron expression to run the job every hour
quarkus.kubernetes.cron-job.schedule=0 * * * *
----

> IMPORTANT: CronJob resources require the https://en.wikipedia.org/wiki/Cron[Cron] expression to specify when to launch the job via the property `quarkus.kubernetes.cron-job.schedule`. If not provide, the build will fail.

Reviewer: ```suggestion
IMPORTANT: CronJob resources require the https://en.wikipedia.org/wiki/Cron[Cron] expression to specify when to launch the job via the property `quarkus.kubernetes.cron-job.schedule`. If not provided, the build will fail.
```

> You can configure the rest of the Kubernetes CronJob configuration using the properties under `quarkus.kubernetes.cron-job.xxx` (see https://quarkus.io/guides/deploying-to-kubernetes#quarkus-kubernetes-kubernetes-config_quarkus.kubernetes.cron-job.parallelism-parallelism[link]).

Reviewer: ```suggestion
You can configure the rest of the Kubernetes CronJob configuration using the properties under `quarkus.kubernetes.cron-job.xxx` (see the https://quarkus.io/guides/deploying-to-kubernetes#quarkus-kubernetes-kubernetes-config_quarkus.kubernetes.cron-job.parallelism-parallelism[Deploying to Kubernetes guide] for more information).
```
